lib/rules/no-unused-vars.js

```diff
@@ -104,6 +139,8 @@ module.exports = {
 			usedIgnoredVar:
 				"'{{varName}}' is marked as ignored but is used{{additional}}.",
 			removeVar: "Remove unused variable '{{varName}}'.",
+			usedOnlyAsType:
+				"'{{varName}}' is {{action}} but only used as a type{{additional}}.",
 		},
 	},
```

Comment:
```suggestion
                },
                
                dialects: ["typescript", "javascript"],
		language: "javascript",
        },
```

Could you add `dialects` and `language` fields to the `meta` property? As far as I know, the newly updated rules that support TypeScript use these fields.

https://github.com/eslint/eslint/blob/28cc7abbb72b29b1cac6fc4253646a7839586064/lib/rules/no-var.js#L202-L203